packages/shared/wallet/src/hooks/use-connect.ts

```diff
@@ -1,89 +1,161 @@
+import { useCallback } from 'react';
 import { trackEvent } from '@notional-finance/helpers';
 import { useNotionalContext } from '@notional-finance/notionable-hooks';
 import { getNetworkFromId, TRACKING_EVENTS } from '@notional-finance/util';
-import { useConnectWallet, useSetChain } from '@web3-onboard/react';
-import { BigNumber, ethers } from 'ethers';
-import { useCallback, useEffect } from 'react';
+import MetaMask from '../images/meta-mask.svg';
+import WalletConnect from '../images/wallet-connect.svg';
+import CoinbaseWallet from '../images/coinbase-wallet.svg';
+import Safe from '../images/safe.svg';
+import { Connector, useConnect, useDisconnect } from 'wagmi';
+import { BigNumber } from 'ethers';
+import { useEffect } from 'react';
+import { providers } from 'ethers';
+import { useMemo } from 'react';
+import type { Account, Chain, Client, Transport } from 'viem';
+import { useAccount, Config, useConnectorClient } from 'wagmi';
 
-export const useConnect = () => {
-  const { globalState, updateNotional } = useNotionalContext();
-  const [
-    { wallet },
-    connect,
-    disconnect,
-    _updateBalances,
-    _setWalletModules,
-    setPrimaryWallet,
-  ] = useConnectWallet();
-  const currentLabel = wallet?.label;
-  const [{ connectedChain }] = useSetChain(currentLabel);
-  const icon = wallet?.icon;
+export function clientToSigner(client: Client<Transport, Chain, Account>) {
+  const { account, chain, transport } = client;
+  const network = {
+    chainId: chain.id,
+    name: chain.name,
+    ensAddress: chain.contracts?.ensRegistry?.address,
+  };
+  const provider = new providers.Web3Provider(transport, network);
+  const signer = provider.getSigner(account.address);
+  return { signer, provider };
+}
 
-  // The first account and chain are considered "selected" by the UI
-  const selectedAddress = globalState?.wallet?.isReadOnlyAddress
-    ? globalState?.wallet?.selectedAddress
-    : wallet?.accounts[0].address;
-  const isReadOnlyAddress = globalState?.wallet?.isReadOnlyAddress;
+/** Hook to convert a Viem Client to an ethers.js Signer. */
+export function useEthersSigner({ chainId }: { chainId?: number } = {}) {
+  const { data: client } = useConnectorClient<Config>({ chainId });
+  return useMemo(() => (client ? clientToSigner(client) : undefined), [client]);
+}
 
-  const onboardChainId = connectedChain?.id;
-  const selectedChain = onboardChainId
-    ? getNetworkFromId(BigNumber.from(onboardChainId).toNumber())
-    : undefined;
+const connectorIcons = {
+  MetaMask: MetaMask,
+  Safe: Safe,
+  WalletConnect: WalletConnect,
+  'Coinbase Wallet': CoinbaseWallet,
+};
+
+export const useWalletConnect = () => {
```

Comment:
Can we put this back to `useConnect` to distinguish from the WalletConnect product